src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/ReferralService.kt

```diff
@@ -71,8 +72,15 @@ class ReferralService(val repository: ReferralRepository) {
       if (referral.serviceCategoryID == null && update.serviceCategoryId == null) {
         errors.add(FieldError(field = "desiredOutcomeIds", error = Code.SERVICE_CATEGORY_MUST_BE_SET))
       }
+
+      // fixme: error if desiredOutcomeIds not valid for service category
+    }
+
+    update.serviceUser?.let {
+      if (referral.serviceUser != null) {
+        errors.add(FieldError(field = "serviceUser", error = Code.FIELD_CANNOT_BE_CHANGED))
```

Comment:
@sldblog your thing on CRN above also makes this check significantly better. 

the CRN cannot be change - the service user details can